sim/druid/talents.go

```diff
@@ -90,6 +91,43 @@ func (druid *Druid) ApplyTalents() {
 	druid.applyPrimalMadness()
 }
 
+func (druid *Druid) applyNaturesGrace() {
+	if druid.Talents.NaturesGrace == 0 {
+		return
+	}
+
+	ngAuraSpellId := []int32{0, 16880, 61345, 61345}[druid.Talents.NaturesGrace]
+	ngAuraSpellHastePct := []float64{0, 0.05, 0.1, 0.15}[druid.Talents.NaturesGrace]
+	ngCooldown := core.Cooldown{
+		Timer:    druid.NewTimer(),
+		Duration: time.Second * 60,
+	}
+
+	ngAura := druid.RegisterAura(core.Aura{
+		Label:    "Natures Grace Proc",
+		ActionID: core.ActionID{SpellID: ngAuraSpellId},
+		Duration: time.Second * 15,
+		Icd:      &ngCooldown,
```

Comment:
The cooldown needs to be attached to `MakeProcTriggerAura` - `Activate` will not check for the ICD again and the trigger has none so you'll just activate it disregarding the cooldown.
Moreover you should add an eclipse callback if the druid has one.
So assign the trigger aura of `MakeProcTriggerAura` to a variable i.E. `ngTrigger` and then you can do something like.

```go
if druid.HasEclipseBar() {
    druid.AddEclipseCallback(func(_ Eclipse, gained bool, _ *core.Simulation) {
        if gained {
			ngTrigger.Icd.Reset()
		}
    }
}
```

Reply:
resolved thanks